pyiron_base/job/core.py

```diff
@@ -429,6 +430,16 @@ def remove_child(self):
         internal function to remove command that removes also child jobs.
         Do never use this command, since it will destroy the integrity of your project.
         """
+        # Check if the job requires to be removed from the full object (This is the case for external Storage)
+        try:
+            requires_full_object = self.get('REQ_OBJ_RM')
+        except ValueError as e:
+            requires_full_object = False
+
+        if requires_full_object:
+            job = self.to_object()
+            job.remove_child()
+
```

Comment:
Is the normal case still necessary after calling `remove_child()` or could it try to delete certain things twice?

Reply:
It should not be needed, since I actually call `super()` as well. However, all remove functions here don't mind if things are not there.

Reply:
I just have some trouble following the control flow here now.  As far as I can tell now

1. `Project.remove_jobs()` inspects jobs and calls `JobPath.remove_child()`.
2. This calls the functions modified here, which loads the job and calls `RDMStorage.remove_child()` to do the actual deletion of the data in S3.
3. But presumably that also calls `super().remove_child()`, which propagates back to `JobPath.remove_child()`, so we go back to 2.

Am I missing where the recursion is broken?  Also because of that `super()` call I'd prefer a 
```python
if requires_full_object:
  ...
else:
  ...
```

Reply:
I am also happy with the if/else. However, I after all have to break the recursion 'by hand'. Since, as long as the `REQUIRE_FULL_OBJ_FOR_RM` Flag is `True` in the hdf, we would loop through the same code over and over again. Therefore I use
```
def remove_child(self):
   remove_s3_storage_access()
   self['REQUIRE_FULL_OBJ_FOR_RM']=False
   super().remove_child()
```
to break this.

Reading my own answer, I think it would be easier to have a `_before_generic_remove_child` method on the job which is called instead of `job.remove_child()`. This breaks the recursion and is much cleaner.

```suggestion
        if requires_full_object:
            job = self.to_object()
            job._before_generic_remove_child()
            
```

Reply:
Sounds good!